packages/dev/core/src/Shaders/ShadersInclude/lightsFragmentFunctions.fx

```diff
@@ -44,9 +44,36 @@ lightingInfo computeLighting(vec3 viewDirectionW, vec3 vNormal, vec4 lightData,
 	return result;
 }
 
-lightingInfo computeSpotLighting(vec3 viewDirectionW, vec3 vNormal, vec4 lightData, vec4 lightDirection, vec3 diffuseColor, vec3 specularColor, float range, float glossiness) {
-	lightingInfo result;
+float getAttenuation(float cosAngle, float exponent) {
+	return max(0., pow(cosAngle, exponent));
+}
+
+float getIESAttenuation(float cosAngle, sampler2D iesLightSampler) {
+	float angle = acos(cosAngle) / PI;

```

Comment:
I think `cosAngle` range is `[0,1]`, so `angle` range is [0, 0.5]: is that expected? Shouldn't we use PI/2 instead of PI?